dask_expr/_merge.py

```diff
@@ -352,6 +353,46 @@ def _lower(self):
         return BlockwiseMerge(left, right, **self.kwargs)
 
     def _simplify_up(self, parent, dependents):
+        if isinstance(parent, Merge):
+            # TODO: Figuring out what we can rewrite and how is some work
+            if parent.how == self.how == "inner":
+                parent = (
+                    (parent.left_on, parent.left, 0)
+                    if parent.right is self
+                    else (parent.right_on, parent.right, 0)
+                )
+                all_frames = [
+                    parent,
+                    (self.left_on, self.left, 1),
+                    (self.right_on, self.right, 2),
+                ]
+
+                def cost(join_tuple):
+                    # This should be a more general cardinality esimate of the
+                    # resulting join but for now just sort them by size
+                    (_, left, _), (_, right, _) = join_tuple
+                    return (right.npartitions, left.npartitions)
```

Comment:
If divisions are known we could actually try to be smart here

Reply:
Cool.  This seems like an easy thing to do.  I'm excited to see if it has impact on TPC-H queries.

Reply:
The implementation looks pleasant too